src/Mimi.jl

```diff
@@ -271,12 +278,25 @@ end
 function connectparameter(m::Model, target_component::Symbol, target_name::Symbol, source_component::Symbol, source_name::Symbol)
     c_target = m.components[target_component]
     c_source = m.components[source_component]
+
+    # Check the units, if provided
+    @assert unitcheck(getmetainfo(m, target_component).parameters[target_name].unit,
+                      getmetainfo(m, source_component).variables[source_name].unit) "Units of $source_component.$source_name do not match $target_component.$target_name."
```

Comment:
I think we should not use ``@assert`` here, there is some discussion that those statements might turn into no-ops for higher optimization levels in julia (JuliaLang/julia#10614). Probably safest to just use an ``if`` and then ``throw`` construction here. I would just throw an ``ErrorException`` if things don't match.